src/MeetingsSDKAdapter.js

```diff
@@ -180,30 +189,59 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
   }
 
   /**
-   * Returns a promise to a local media stream based on the given constraints.
+   * Returns a promise to a local media stream based on the given mediaDirection, audioVideo options.
    *
    * @see {@link MediaStream|https://developer.mozilla.org/en-US/docs/Web/API/MediaStream}.
    *
    * @private
    * @param {string} ID ID of the meeting for which to fetch streams
-   * @param {object} constraint Media stream constraints
+   * @param {object} mediaDirection A configurable options object for joining a meetings
+   * @param {object} audioVideo audio/video object to set audioinput and videoinput devices
    * @returns {Promise.<MediaStream>} Promise to requested media stream
    */
-  async getStream(ID, constraint) {
+  async getStream(ID, mediaDirection, audioVideo) {
     let localStream = null;
 
     try {
       const sdkMeeting = this.fetchMeeting(ID);
 
-      [localStream] = await sdkMeeting.getMediaStreams(constraint);
+      [localStream] = await sdkMeeting.getMediaStreams(mediaDirection, audioVideo);
     } catch (error) {
       // eslint-disable-next-line no-console
-      console.error(`Unable to retrieve local media stream for meeting "${ID}" with constraint "${constraint}"`, error);
+      console.error('Unable to retrieve local media stream for meeting', ID, 'with mediaDirection', mediaDirection, 'and audioVideo', audioVideo, 'reason:', error);
     }
 
     return localStream;
   }
 
+  /**
+   * Returns available media devices.
+   *
+   * @param {string} ID ID of the meeting
+   * @param {'videoinput'|'audioinput'|'audiooutput'} type  String specifying the device type.
+   * See {@link https://developer.mozilla.org/en-US/docs/Web/API/MediaDeviceInfo/kind|MDN}
+   * @returns {MediaDeviceInfo[]|null} Array containing media devices or null if devices can't be read.
```

Comment:
it's ok to have now just "Array containing media devices"

null is no longer returned in case of an error. You can even remove the `= null` assignment from line 228.